### PR TITLE
fix: add timeout in stop command

### PR DESCRIPTION
**Pull request by mozhuli**
Add timeout in stop command

Signed-off-by: mozhuli <21621232@zju.edu.cn>

**1.Describe what this PR did**

**2.Does this pull request fix one issue?** 

**3.Describe how you did it**

**4.Describe how to verify it**

**5.Special notes for reviews**

### CODE REVIEW

**Comment by allencloud** [comment]
Thanks a lot for your work. It is so good. To be honest, we still have one thing missing.
Could you first to add this APi `POST /containers/{id}/stop` in swagger.yml as a previous PR?
Then we can move on this to check if it is reasonable. @mozhuli

File: ctrd/container.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func (c *Client) DestroyContainer(ctx context.Context, id string) (*Message, err
 	}
 
 	waitExit := func() *Message {
-		return c.ProbeContainer(ctx, id, time.Second*5)
+		return c.ProbeContainer(ctx, id, time.Duration(timeout)*time.Second)
```
Comment (allencloud) [inline ctrd/container.go]: We can directly use `timeout * time.Second`, right?

Comment (allencloud) [inline ctrd/container.go]: We can directly use `timeout * time.Second`, right?

Comment (mozhuli) [reply ctrd/container.go]: right

**Comment by mozhuli** [comment]
@allencloud The API which has no more other API structs still need to define in the swagger.yml? if so, I can help with listing the APIs which have not define in the swagger.yml and adding them all in another PR.

**Comment by allencloud** [comment]
> The API which has no more other API structs still need to define in the swagger.yml?

Yes. We need this endpoint to auto generate doc https://github.com/alibaba/pouch/blob/master/docs/api/api.md 😄 
@mozhuli

**Comment by mozhuli** [comment]
@allencloud as the new test package merged, I will rebase and add stop test case later

**Comment by allencloud** [comment]
That could not be better. Thanks

**Comment by mozhuli** [comment]
@allencloud @0x04C2  would you have a look at the CI logs, it seems that the test `PruneAllContainers` method has some problems?

**Comment by 0x04C2** [comment]
Yes. @mozhuli . I'm still checking. :)

**Comment by allencloud** [comment]
Please rebase this PR to the latest master. 😄

**Comment by mozhuli** [comment]
@allencloud   rebaesd! and the default timeout is 10 seconds which same with docker, I think it's too long, should we lower it?

File: ctrd/container.go (hunk 1 above)
Comment (allencloud) [reply ctrd/container.go]: Do we have a plan to change this? 😄

File: apis/swagger.yml [hunk 2]
```diff
@@ -337,6 +337,41 @@ paths:
...
+          in: "query"
+          description: "Number of seconds to wait before killing the container"
+          type: "integer"
+      responses:
+        204:
+          description: "no error"
+        304:
+          description: "container already stopped"
+          schema:
+            $ref: "#/definitions/Error"
```
Comment (allencloud) [inline apis/swagger.yml]: I know that this is from Moby's swagger.yml. While I am afraid status code 3xx is not an error. 
We can have a discussion that whether it is proper to set 304 here? 

We can take "pause an paused container" as an example. WDYT?

Comment (mozhuli) [reply apis/swagger.yml]: OK! I will update

**Comment by mozhuli** [comment]
@allencloud updated!

File: ctrd/container.go (hunk 1 above)
Comment (mozhuli) [reply ctrd/container.go]: it seems that we must use `time.Duration()` when we use variable，we can use this way if we use the constant.

File: cli/stop.go [hunk 3]
```diff
@@ -39,11 +39,15 @@ func (s *StopCommand) addFlags() {
 
 // runStop is the entry of stop command.
 func (s *StopCommand) runStop(args []string) error {
+	timeout := "0"
 	apiClient := s.cli.Client()
 
 	container := args[0]
+	if len(args) == 2 {
+		timeout = args[1]
```
Comment (allencloud) [inline cli/stop.go]: I think we should add this timeout into flag, not args,like:
```
flagSets.IntVarP(&cc.timeout, "time", "t", 10, "Seconds to wait for stop before killing it")
```

Comment (mozhuli) [reply cli/stop.go]: It is better, I will update it

**Comment by allencloud** [comment]
LGTM
maybe in the future, we should change `Args:  cobra.MinimumNArgs(1),` into `Args: cobra.ExactArgs(1)`.

**Pull request merged**